docs/data_types/tensor/tensor.md

```diff
@@ -0,0 +1,224 @@
+
+```python
+doc = MyDoc(tensor=np.zeros((100, 100)))
+``` 
+
+### Axes validation
+
+You can check that the number of axis is correct as well `NdArray['x','y']`, `TorchTensor['x','y']`, `TensorFlowTensor['x','y']`.
+
+```python
+from docarray import BaseDoc
+from docarray.typing import NdArray
+
+
+class MyDoc(BaseDoc):
+    tensor: NdArray['x', 'y']
+``` 
+
+Here you can only pass a tensor with two axis. `np.zeros(10, 12)` will work but `np.zeros(10, 12, 3)` will raise an error.
+
+### Axis names
+
+You can as well specify that two axis should have the same dimension with the syntax `NdArray['x', 'x']`, `TorchTensor['x', 'x']`, `TensorFlowTensor['x', 'x']`.
+
+```python
+from docarray import BaseDoc
+from docarray.typing import NdArray
+
+
+class MyDoc(BaseDoc):
+    tensor: NdArray['x', 'x']
+``` 
+
+Here you can only pass a tensor with two axis with the same dimension. `np.zeros(10, 10)` will work but `np.zeros(10, 12)` will raise an error.
+
+### Arbitrary number of axis
+
+You can specify that your shape can have an arbitrary number of axis with the syntax `NdArray['x', ...]`, or `NdArray[..., 'x']`
+
+```python
+from docarray import BaseDoc
+from docarray.typing import NdArray
+
+
+class MyDoc(BaseDoc):
+    tensor: NdArray[100, ...]
+``` 
+
+Here you can only pass a tensor with at least one axis with dimension 100. `np.zeros(100, 10)` will work but `np.zeros(10, 12)` will raise an error.
+
+## Tensor type validation
+
+You don't need to directly instantiate the  [`NdArray`][docarray.typing.tensor.NdArray] , [`TorchTensor`][docarray.typing.tensor.TorchTensor], or [`TensorFlowTensor`][docarray.typing.tensor.TensorFlowTensor] by yourself.
+
+Instead, you should use them as type hints on [`BaseDoc`][docarray.base_doc.doc.BaseDoc] fields, where they perform data validation.
+During this process, [`BaseDoc`][docarray.base_doc.doc.BaseDoc] will cast the native tensor type into the respective DocArray tensor type.
+
+Let's look at an example:
+
+```python
+from docarray import BaseDoc
+from docarray.typing import NdArray
+
+import numpy as np
+
+
+class MyDoc(BaseDoc):
+    tensor: NdArray
+
+
+doc = MyDoc(tensor=np.zeros(100))
+
+assert isinstance(doc.tensor, NdArray)  # True
+``` 
+Here you see that the `doc.tensor` is an `NdArray`:
+
+```python
+assert isinstance(doc.tensor, np.ndarray)  # True as well
+``` 
+
+But since it inherits from `np.ndarray`, you can also use it as a normal NumPy array. The same holds for PyTorch and `TorchTensor`.
+
+## Type coercion with different tensor types 
+
+
+DocArray also supports type coercion between different tensor types. This mean that if you pass a different tensor type to a tensor field, it will be converted to the right tensor type.
+
+For instance if you define a field of type [`TorchTensor`][docarray.typing.tensor.TorchTensor] and you pass a Numpy array to it, it will be converted to a [`TorchTensor`][docarray.typing.tensor.TorchTensor].
```

Comment:
```suggestion
For instance, if you define a field of type [`TorchTensor`][docarray.typing.tensor.TorchTensor] and you pass a NumPy array to it, it will be converted to a [`TorchTensor`][docarray.typing.tensor.TorchTensor].
```